opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/HistoryController.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ */
+
+package org.opengrok.web.api.v1.controller;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.jetbrains.annotations.NotNull;
+import org.opengrok.indexer.configuration.RuntimeEnvironment;
+import org.opengrok.indexer.history.History;
+import org.opengrok.indexer.history.HistoryEntry;
+import org.opengrok.indexer.history.HistoryException;
+import org.opengrok.indexer.history.HistoryGuru;
+import org.opengrok.indexer.web.messages.JSONable;
+import org.opengrok.web.api.v1.filter.CorsEnable;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.SortedSet;
+
+@Path("/history")
+public class HistoryController {
+
+    private RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+
+    private static final int MAX_RESULTS = 1000;
+
+    static class HistoryEntryDTO implements JSONable {
+        @JsonProperty
+        private String revision;
+        @JsonProperty
+        private Date date;
+        @JsonProperty
+        private String author;
+        @JsonProperty
+        private String tags;
+        @JsonProperty
+        private String message;
+        @JsonProperty
+        private SortedSet<String> files;
```

Comment:
all can be final

Reply:
they would have to be initialized in the plain constructor first I guess.

Reply:
I get it, it's the serialization issue